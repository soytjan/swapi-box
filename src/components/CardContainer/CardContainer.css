@import url("https://fonts.googleapis.com/css?family=Bungee+Hairline|Bungee+Outline|Roboto|Roboto+Condensed");
.card-container {
  display: grid;
  grid-template-columns: repeat(6, 1fr);
  grid-gap: 20px;
  background-color: #2B2B2D; }
  .card-container h2 {
    grid-column: 1/7;
    text-transform: capitalize;
    font-family: "Bungee Hairline", cursive;
    font-size: 1.5em;
    color: white; }

.loading {
  grid-column: 1/7; }
