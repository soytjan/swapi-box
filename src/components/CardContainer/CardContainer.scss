@import "../../variables.scss";

.card-container {
  display: grid;
  grid-template-columns: repeat(6, 1fr);
  grid-gap: 20px;
  background-color: $box-bg-color;

  h2 {
    grid-column: 1/7;
    text-transform: capitalize;
    font-family: $header-box-font;
    font-size: 1.5em;
    color: white;
  }
}