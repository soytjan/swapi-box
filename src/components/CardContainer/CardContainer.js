import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';
import './CardContainer.css'

const CardContainer = ({data, type, onFavClick, favorites}) => {
  let renderedCards;
  
  if (type === 'favorites' && data.length === 0) {
    renderedCards = (<p className='loading'>You have no favorites selected!</p>)
  } else if (data.length === 0) {
    renderedCards = (<div className='loading'>Loading...</div>)
  } else {
    renderedCards = data.map((info, index) => {
      const isFavorited = favorites.includes(info) ? 'favorited' : '';
    
      return <Card info={info} type={type} isFavorited={isFavorited} onFavClick={onFavClick} key={`${index}${Date.now()}`} />
    })
  }

  return(
    <section className="card-container">
      <h2>{type}</h2>
      {renderedCards}
    
    </section>
  )
}

CardContainer.propTypes = {
  data: PropTypes.array,
  type: PropTypes.string,
  onFavClick: PropTypes.func,
  favorites: PropTypes.array,
};

export default CardContainer;