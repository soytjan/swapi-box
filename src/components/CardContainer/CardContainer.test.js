/* eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer';

describe('CardContainer', () => {
  let mockFavClick;
  let mockData;

  beforeAll(() => {
    mockFavClick = jest.fn();
    mockData = [
      {name: "Coruscant", 
      terrain: "cityscape, mountains", 
      population: "1000000000000", 
      climate: "temperate", 
      residents: ["Finis Valorum", "Adi Gallia", "Jocasta Nu"]}
    ];

  })

  it('should match snapshot', () => {
    const renderedComponent = shallow(<CardContainer swData={mockData} type='planets' onFavClick={mockFavClick} favorites={[]} />)

    expect(renderedComponent).toMatchSnapshot();
  });

  it('should have Loading page if no data is passed in', () => {
    const renderedComponent = shallow(<CardContainer swData={[]} type='planets' onFavClick={mockFavClick} favorites={[]} />)

    expect(renderedComponent).toMatchSnapshot();
  });

  it('when there are no favorites, should have a message indicating there are no favorites', () => {
    const renderedComponent = shallow(<CardContainer swData={[]} type='favorites' onFavClick={mockFavClick} favorites={[]} />)

    expect(renderedComponent).toMatchSnapshot();
  });
})