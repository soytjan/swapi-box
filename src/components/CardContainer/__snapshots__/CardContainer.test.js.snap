// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should have Loading page if no data is passed in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    favorites={Array []}
    onFavClick={[Function]}
    swData={Array []}
    type="planets"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          planets
</h2>,
        <div
          className="loading"
>
          Loading...
</div>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "planets",
        },
        "ref": null,
        "rendered": "planets",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
          "className": "loading",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            planets
</h2>,
          <div
            className="loading"
>
            Loading...
</div>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "planets",
          },
          "ref": null,
          "rendered": "planets",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
            "className": "loading",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CardContainer should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    favorites={Array []}
    onFavClick={[Function]}
    swData={
        Array [
            Object {
              "climate": "temperate",
              "name": "Coruscant",
              "population": "1000000000000",
              "residents": Array [
                "Finis Valorum",
                "Adi Gallia",
                "Jocasta Nu",
              ],
              "terrain": "cityscape, mountains",
            },
          ]
    }
    type="planets"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          planets
</h2>,
        Array [
          <Card
            info={
                        Object {
                                    "climate": "temperate",
                                    "name": "Coruscant",
                                    "population": "1000000000000",
                                    "residents": Array [
                                      "Finis Valorum",
                                      "Adi Gallia",
                                      "Jocasta Nu",
                                    ],
                                    "terrain": "cityscape, mountains",
                                  }
            }
            isFavorited=""
            onFavClick={[Function]}
            type="planets"
/>,
        ],
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "planets",
        },
        "ref": null,
        "rendered": "planets",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "01517859415283",
        "nodeType": "function",
        "props": Object {
          "info": Object {
            "climate": "temperate",
            "name": "Coruscant",
            "population": "1000000000000",
            "residents": Array [
              "Finis Valorum",
              "Adi Gallia",
              "Jocasta Nu",
            ],
            "terrain": "cityscape, mountains",
          },
          "isFavorited": "",
          "onFavClick": [Function],
          "type": "planets",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            planets
</h2>,
          Array [
            <Card
              info={
                            Object {
                                          "climate": "temperate",
                                          "name": "Coruscant",
                                          "population": "1000000000000",
                                          "residents": Array [
                                            "Finis Valorum",
                                            "Adi Gallia",
                                            "Jocasta Nu",
                                          ],
                                          "terrain": "cityscape, mountains",
                                        }
              }
              isFavorited=""
              onFavClick={[Function]}
              type="planets"
/>,
          ],
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "planets",
          },
          "ref": null,
          "rendered": "planets",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "01517859415283",
          "nodeType": "function",
          "props": Object {
            "info": Object {
              "climate": "temperate",
              "name": "Coruscant",
              "population": "1000000000000",
              "residents": Array [
                "Finis Valorum",
                "Adi Gallia",
                "Jocasta Nu",
              ],
              "terrain": "cityscape, mountains",
            },
            "isFavorited": "",
            "onFavClick": [Function],
            "type": "planets",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CardContainer when there are no favorites, should have a message indicating there are no favorites 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    favorites={Array []}
    onFavClick={[Function]}
    swData={Array []}
    type="favorites"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          favorites
</h2>,
        <p
          className="loading"
>
          You have no favorites selected!
</p>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "favorites",
        },
        "ref": null,
        "rendered": "favorites",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You have no favorites selected!",
          "className": "loading",
        },
        "ref": null,
        "rendered": "You have no favorites selected!",
        "type": "p",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            favorites
</h2>,
          <p
            className="loading"
>
            You have no favorites selected!
</p>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "favorites",
          },
          "ref": null,
          "rendered": "favorites",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You have no favorites selected!",
            "className": "loading",
          },
          "ref": null,
          "rendered": "You have no favorites selected!",
          "type": "p",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
